core/src/main/java/com/google/cloud/sql/ConnectorRegistry.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql;
+
+import com.google.cloud.sql.core.CoreSocketFactory;
+
+/** The public java API to configure the connector registry. */
+public enum ConnectorRegistry {
```

Comment:
The name is nice, but maybe we should just make CoreSocketFactory.getInstance() public?

Reply:
Alternatively, we could do this (which seems more keeping with the existing interface):

``` java
CoreSocketFactory.register("my-connection", config);
```

Reply:
Let's discuss some design options.

Reply:
My vote is for:

```
CoreSocketFactory.register("my-connection", config);
```